backend/src/common/util/workflow.go

```diff
@@ -200,6 +200,19 @@ func (w *Workflow) SetAnnotationsToAllTemplates(key string, value string) {
 	}
 }
 
+// SetLabels sets labels on all templates in a Workflow
+func (w *Workflow) SetLabelsToAllTemplates(key string, value string) {
+	if len(w.Spec.Templates) == 0 {
+		return
+	}
+	for index, _ := range w.Spec.Templates {
+		if w.Spec.Templates[index].Metadata.Labels == nil {
+			w.Spec.Templates[index].Metadata.Labels = make(map[string]string)
+		}
+		w.Spec.Templates[index].Metadata.Labels[key] = value
```

Comment:
should this code skip setting the cache value if it exists? This way it will respect user's input if user disable the cache for a given step.

Reply:
Yes. The cache server will only receive the pod with labels: cache_enabled = true. In the future we can have UI support with toggling cache behavior for a run with this label. Per step cache behavior control will be base on MaxCacheStaleness.

Reply:
> should this code skip setting the cache value if it exists? This way it will respect user's input if user disable the cache for a given step.

Per-step cache disabling is performed using a different mechanism (setting task.max_cache_staleness = "0").

I'm OK with not changing the value if it already exists. But it should not exist since it's a vendored label and SDK does not set it.

Reply:
I'm a bit confused. Do we have two flags to control the cache for a given step, cache_enabled and max_cache_staleness? if one supersets the other, we should consider dedup them. 

If cache_enabled is not exposed to end user, what's the purpose of having it? Should it be a feature flag at the caching server side, instead of at the workflow side?